Plugins/AWSLambdaPackager/Plugin.swift

```diff
@@ -0,0 +1,344 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftAWSLambdaRuntime open source project
+//
+// Copyright (c) 2022 Apple Inc. and the SwiftAWSLambdaRuntime project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftAWSLambdaRuntime project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import Dispatch
+import Foundation
+import PackagePlugin
+
+#if canImport(Glibc)
+import Glibc
+#endif
+
+@main
+struct AWSLambdaPackager: CommandPlugin {
+    func performCommand(context: PackagePlugin.PluginContext, arguments: [String]) async throws {
+        let configuration = Configuration(context: context, arguments: arguments)
+        guard !configuration.products.isEmpty else {
+            throw Errors.unknownProduct("no appropriate products found to package")
+        }
+
+        #if os(macOS)
+        let builtProducts = try self.buildInDocker(
+            packageIdentity: context.package.id,
+            packageDirectory: context.package.directory,
+            products: configuration.products,
+            toolsProvider: { name in try context.tool(named: name).path },
+            outputDirectory: configuration.outputDirectory,
+            baseImage: configuration.baseImage,
+            buildConfiguration: configuration.buildConfiguration,
+            verboseLogging: configuration.verboseLogging
+        )
+        #elseif os(Linux)
+        let builtProducts = try self.build(
+            products: configuration.products,
+            buildConfiguration: configuration.buildConfiguration,
+            verboseLogging: configuration.verboseLogging
+        )
+        #else
```

Comment:
Not sure this is the right condition... What happens if I am on Ubuntu Linux?

Reply:
good point. do we want to check for distro?

Reply:
I guess we have to...